integration-cli/docker_utils.go

```diff
@@ -396,6 +396,55 @@ func deleteAllContainers() error {
 	return nil
 }
 
+var protectedImages = map[string]struct{}{}
+
+func init() {
+	out, err := exec.Command(dockerBinary, "images").CombinedOutput()
+	if err != nil {
+		panic(err)
+	}
+	lines := strings.Split(string(out), "\n")[1:]
+	for _, l := range lines {
+		if l == "" {
+			continue
+		}
+		fields := strings.Fields(l)
+		imgTag := fields[0] + ":" + fields[1]
+		protectedImages[imgTag] = struct{}{}
```

Comment:
I guess this would technically add `<none>:<none>` to the `protectedImages` hash, meaning that any new dangling images created would be kept too, so this will just balloon instead.